src/ciadmin/check/check_grants.py

```diff
@@ -44,6 +44,88 @@ async def check_grant_aliases():
     assert not unknown_aliases
 
 
+@pytest.mark.asyncio
+async def check_grant_pools(generate_resources):
+    """
+    Ensures that we don't grant things for non-existent worker pools.
+    """
+    generated = await generate_resources("worker_pools")
+
+    # Known scopes that reference worker-pools.
+    prefixes = {
+        "generic-worker:allow-rdp:",
+        "generic-worker:os-group:",
+        "generic-worker:run-as-administrator:",
+        "queue:create-task:",
+        "queue:quarantine-worker:",
+    }
+    # These providers are not managed by worker-manager, so valid pools can't
+    # be detected.
+    ignore_providers = {
+        "bitbar",
+        "built-in",
+        "lambda",
+        "performance-hardware",
+        "proj-autophone",
+        "releng-hardware",
+        "scriptworker-k8s",
+        "scriptworker-prov-v1",
+    }
+
+    # We validate the raw grants rather than the generated grants to allow for
+    # things like `{trust_domain}-t/*`. This will inevitably generate scopes
+    # that don't reference valid pools, but that's ok as the intent of this
+    # check is to keep grants.yml clean, not the generated grants.
+    grants = await Grant.fetch_all()
+    pools = [p.workerPoolId for p in generated.filter("WorkerPool=.*")]
+    invalid_scopes = set()
+
+    for grant in grants:
+        for scope in grant.scopes:
+            assert isinstance(scope, str)
+
+            if not any(scope.startswith(p) for p in prefixes):
```

Comment:
```suggestion
            if not scope.startswith(tuple(prefixes)):
```
(or make prefixes a tuple and `...startswith(prefixes)`)

Reply:
I always forget that `startswith` can do that.